openlogprobs/models.py

```diff
@@ -108,28 +111,67 @@ def argmax(self, prefix: str, logit_bias:  Dict[str, float] = {}) -> int:
         return enc.encode(output)[0]
     
     def topk(self, prefix: str, logit_bias: Dict[str, float] = {}) -> Dict[int, float]:
-        enc = self.encoding
-        model = self.model
-        system = self.system
-        if model == "gpt-3.5-turbo-instruct":
-            if logit_bias is not None:
-                response = self.client.completions.create(
-                    model=model,
-                    prompt=prefix,
-                    temperature=1,
-                    max_tokens=1,
-                    logit_bias=logit_bias,
-                    logprobs=5,
-                )
-            else:
-                response = self.client.completions.create(
-                    model=model,
-                    prompt=prefix,
-                    temperature=1,
-                    max_tokens=1,
-                    logprobs=5,
-                )
+        for i in range(self.max_api_retries):
```

Comment:
We should handle this with the openai client: https://github.com/openai/openai-python/blob/main/src/openai/_client.py#L74

Reply:
I didn't see a backoff strategy in the openai client, and it looks like they recommend using other libraries (or implementing from scratch) to do retries with backoff here:

https://cookbook.openai.com/examples/how_to_handle_rate_limits#retrying-with-exponential-backoff

Would you be down for using the `tenacity` or `backoff` libraries they mention in that cookbook link, or even keeping this custom implementation?

Reply:
I've used and like the tenacity library -- I thought after the refactor of the openai python client library, they recommended not using external libraries for retrying. Do you know whether the cookbook is outdated, or how outdated?

Reply:
I looked around a bit more, and looks like the same rate limit guide is in the API documentation itself:

https://platform.openai.com/docs/guides/rate-limits?context=tier-free

The final call is yours on whether to keep a manual backoff strategy or just increase max_retries in the openai client, it should be fine either way!